x/mongo/driver/topology/pool_test.go

```diff
@@ -440,7 +440,16 @@ func TestPool(t *testing.T) {
 			_, err = p.checkOut(context.Background())
 			var want error = ConnectionError{Wrapped: dialErr, init: true}
 			assert.Equalf(t, want, err, "should return error from calling checkOut()")
-			assert.Equalf(t, 0, p.totalConnectionCount(), "pool should have 0 total connections")
+			// If a connection initialization error occurs during checkOut, removing and closing the
+			// failed connection happen asynchronously with the checkOut. Wait for up to 10s for the
```

Comment:
```suggestion
			// failed connection happens asynchronously with the checkOut. Wait for up to 10s for the
```
10 seconds seems like a while? Would 100 ticks pass before the connection is removed + closed?

Reply:
That's a good question. If the connection being removed and closed actually took 10s to complete, that would be quite unusual but technically valid behavior. However, new connection requests could be blocked (e.g. if the pool is at `maxPoolSize`) until the connection is removed, so I'll consider changing it to something shorter like 2s.

As far as "happen" vs "happens", I actually think "happen" is correct since "removing" and "closing" both happen asynchronously with the checkOut. I can add "both" to the comment make the sentence sound better.

Reply:
2s sounds good to me; thanks! And my bad, I just misread the comment. The addition of "both" seems good.